orderbook/src/api/get_fee_and_quote.rs

```diff
@@ -1,21 +1,47 @@
-use crate::{api::price_estimation_error_to_warp_reply, fee::MinFeeCalculating};
-use anyhow::{anyhow, Result};
+use crate::api::{price_estimation_error_to_warp_reply, WarpReplyConverting};
+use crate::{
+    api::{
+        order_validation::OrderValidating,
+        post_quote::{response, OrderQuoteRequest, OrderQuoteResponse, OrderQuoteSide, SellAmount},
+    },
+    fee::MinFeeCalculating,
+};
+use anyhow::Result;
 use chrono::{DateTime, Utc};
 use ethcontract::{H160, U256};
-use model::h160_hexadecimal;
-use model::{order::OrderKind, u256_decimal};
+use model::{h160_hexadecimal, u256_decimal};
 use serde::{Deserialize, Serialize};
-use shared::price_estimation::{self, PriceEstimating, PriceEstimationError};
-use std::convert::Infallible;
-use std::sync::Arc;
-use warp::{hyper::StatusCode, reply, Filter, Rejection, Reply};
+use shared::price_estimation::{PriceEstimating, PriceEstimationError};
+use std::{convert::Infallible, sync::Arc};
+use warp::{hyper::StatusCode, reply::Json, Filter, Rejection, Reply};
 
-#[derive(Serialize)]
+#[derive(Debug, Serialize)]
 #[serde(rename_all = "camelCase")]
-struct Fee {
+pub struct Fee {
     #[serde(with = "u256_decimal")]
-    amount: U256,
-    expiration_date: DateTime<Utc>,
+    pub amount: U256,
+    pub expiration_date: DateTime<Utc>,
+}
+
+#[derive(Debug)]
+pub enum FeeError {
```

Comment:
This should be in the `post_quote` module seeing as its only ever created there in a private function.